src/neo-vm/ExecutionEngine.cs

```diff
@@ -255,39 +554,63 @@ private void ExecuteOp(OpCode opcode, ExecutionContext context)
                                 return;
                             }
                             context.EvaluationStack.Push(context.EvaluationStack.Peek(n));
+                            break;
                         }
-                        break;
                     case OpCode.ROLL:
                         {
                             int n = (int)context.EvaluationStack.Pop().GetBigInteger();
+
                             if (n < 0)
                             {
                                 State |= VMState.FAULT;
                                 return;
                             }
                             if (n == 0) break;
+
                             context.EvaluationStack.Push(context.EvaluationStack.Remove(n));
+                            CheckStackSize(false, -1);
```

Comment:
`CheckStackSize` should be before `if (n == 0) break;` and should be `CheckStackSize(true, -1)`.